packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewFlutterPlugin.java

```diff
@@ -38,36 +50,100 @@ public WebViewFlutterPlugin() {}
+    InstanceManager instanceManager = new InstanceManager();
+    viewRegistry.registerViewFactory(
+        "plugins.flutter.io/webview", new FlutterWebViewFactory(instanceManager));
+    WebViewHostApi.setup(
+        binaryMessenger,
+        new WebViewHostApiImpl(
+            instanceManager, new WebViewHostApiImpl.WebViewProxy(), activity, containerView));
+    WebViewClientHostApi.setup(
+        binaryMessenger,
+        new WebViewClientHostApiImpl(
+            instanceManager,
+            new WebViewClientHostApiImpl.WebViewClientCreator(),
+            new WebViewClientFlutterApiImpl(binaryMessenger, instanceManager)));
+    WebChromeClientHostApi.setup(
+        binaryMessenger,
+        new WebChromeClientHostApiImpl(
+            instanceManager,
+            new WebChromeClientHostApiImpl.WebChromeClientCreator(),
+            new WebChromeClientFlutterApiImpl(binaryMessenger, instanceManager)));
+    DownloadListenerHostApi.setup(
+        binaryMessenger,
+        new DownloadListenerHostApiImpl(
+            instanceManager,
+            new DownloadListenerHostApiImpl.DownloadListenerCreator(),
+            new DownloadListenerFlutterApiImpl(binaryMessenger, instanceManager)));
+    JavaScriptChannelHostApi.setup(
+        binaryMessenger,
+        new JavaScriptChannelHostApiImpl(
+            instanceManager,
+            new JavaScriptChannelHostApiImpl.JavaScriptChannelCreator(),
+            new JavaScriptChannelFlutterApiImpl(binaryMessenger, instanceManager),
+            new Handler(activity.getMainLooper())));
+    WebSettingsHostApi.setup(
+        binaryMessenger,
+        new WebSettingsHostApiImpl(
+            instanceManager, new WebSettingsHostApiImpl.WebSettingsCreator()));
+    new FlutterCookieManager(binaryMessenger);
+  }
+
   @Override
   public void onAttachedToEngine(FlutterPluginBinding binding) {
-    BinaryMessenger messenger = binding.getBinaryMessenger();
-    binding
-        .getPlatformViewRegistry()
-        .registerViewFactory(
-            "plugins.flutter.io/webview",
-            new FlutterWebViewFactory(messenger, /*containerView=*/ null));
-    flutterCookieManager = new FlutterCookieManager(messenger);
+    this.pluginBinding = binding;
+    flutterCookieManager = new FlutterCookieManager(binding.getBinaryMessenger());
   }
 
   @Override
-  public void onDetachedFromEngine(FlutterPluginBinding binding) {
+  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     if (flutterCookieManager == null) {
       return;
     }
 
     flutterCookieManager.dispose();
     flutterCookieManager = null;
   }
+
+  @Override
+  public void onAttachedToActivity(@NonNull ActivityPluginBinding activityPluginBinding) {
+    setUp(
+        pluginBinding.getBinaryMessenger(),
+        pluginBinding.getPlatformViewRegistry(),
+        activityPluginBinding.getActivity(),
+        null);
+  }
+
+  @Override
+  public void onDetachedFromActivityForConfigChanges() {}
```

Comment:
The activity was passed to `WebViewHostApi`; doesn't that need to be disconnected somehow at this point?

Reply:
I switch the context for `WebViewHostApi` to `FlutterPluginBinding.applicationContext()` for the explanation I gave above for `onAttachedToActivity`.